test/microstrain_test.cpp

```diff
@@ -5,24 +5,43 @@
 
 namespace detail
 {
-    void warn_c_strings_equal(const char* string1, const char* string2)
+#ifdef MICROSTRAIN_TEST_USE_DOCTEST
+    void check_c_strings_equal(const char* actual, const char* expected)
     {
-        INFO(std::string(string1));
-        INFO(std::string(string2));
-        WARN_EQ(strcmp(string1, string2), 0);
+        const size_t safe_actual_size = strnlen(actual, MAX_STRING_LENGTH);
+        const size_t safe_expected_size = strnlen(expected, MAX_STRING_LENGTH);
```

Comment:
You'd need to use this constant everywhere the buffer is declared, i.e. `buffer[MAX_STRING_LENGTH]` instead of `buffer[1024]`.

I was originally against a global definition since it technically restricts the test design, and requires the test to honor this convention, but I don't think it's a serious limitation.